pkg/authorization/api/register.go

```diff
@@ -1,34 +1,52 @@
 package api
 
 import (
+	kapi "k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/api/unversioned"
 	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/watch/versioned"
 )
 
-const GroupName = ""
-const FutureGroupName = "authorization.openshift.io"
+const (
+	LegacyGroupName = ""
+	GroupName       = "authorization.openshift.io"
+)
+
+var (
+	SchemeGroupVersion       = unversioned.GroupVersion{Group: GroupName, Version: runtime.APIVersionInternal}
+	LegacySchemeGroupVersion = unversioned.GroupVersion{Group: LegacyGroupName, Version: runtime.APIVersionInternal}
+
+	LegacySchemeBuilder    = runtime.NewSchemeBuilder(addLegacyKnownTypes)
+	AddToSchemeInCoreGroup = LegacySchemeBuilder.AddToScheme
 
-// SchemeGroupVersion is group version used to register these objects
-var SchemeGroupVersion = unversioned.GroupVersion{Group: GroupName, Version: runtime.APIVersionInternal}
+	SchemeBuilder = runtime.NewSchemeBuilder(addKnownTypes)
+	AddToScheme   = SchemeBuilder.AddToScheme
+)
 
 // Kind takes an unqualified kind and returns back a Group qualified GroupKind
 func Kind(kind string) unversioned.GroupKind {
 	return SchemeGroupVersion.WithKind(kind).GroupKind()
 }
 
+// LegacyKind takes an unqualified kind and returns back a Group qualified GroupKind
+func LegacyKind(kind string) unversioned.GroupKind {
+	return LegacySchemeGroupVersion.WithKind(kind).GroupKind()
+}
+
 // Resource takes an unqualified resource and returns back a Group qualified GroupResource
 func Resource(resource string) unversioned.GroupResource {
 	return SchemeGroupVersion.WithResource(resource).GroupResource()
 }
 
-var (
-	SchemeBuilder = runtime.NewSchemeBuilder(addKnownTypes)
-	AddToScheme   = SchemeBuilder.AddToScheme
-)
+// LegacyResource takes an unqualified resource and returns back a Group qualified
+// GroupResource using legacy API
+func LegacyResource(resource string) unversioned.GroupResource {
+	return LegacySchemeGroupVersion.WithResource(resource).GroupResource()
+}
 
 // Adds the list of known types to api.Scheme.
 func addKnownTypes(scheme *runtime.Scheme) error {
-	scheme.AddKnownTypes(SchemeGroupVersion,
+	types := []runtime.Object{
```

Comment:
@liggitt guess it is easier to manage what are types that really belongs to the package/group vs. the meta types. @sttts ?